src/datepicker/datepicker.js

```diff
@@ -811,6 +813,9 @@ function(scope, element, attrs, $compile, $parse, $document, $rootScope, $positi
 
     if (angular.isString(viewValue)) {
       var date = dateParser.parse(viewValue, dateFormat, scope.date);
+      for(var i = 0; isNaN(date) && i < altInputFormats.length; i++) {
```

Comment:
Would it make more sense to move the isNaN check to inside the loop after the dateParser is run? That way one can just `break` out of the loop if it is a valid date.